### PR TITLE
fixes bug 1201270 - Migration to backfill last 28 days of signature_summary_os

### PR DESCRIPTION
**Pull request by peterbe**
@twobraids r?

### CODE REVIEW

**Comment by twobraids** [comment]
looks like you may have to run both stored procs?  one to ensure the partition exists, the other to fill it with data?

**Comment by peterbe** [comment]
Ok. Pushed another attempt to fix that.

**Comment by peterbe** [comment]
Seems happier now.

File: alembic/versions/16b2bee7db72_backfill_signature_summary_os_28_days.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    # We only ever, at most, show 28 days in the report list that
+    # shows this signature summary.
+    today = datetime.datetime.utcnow().date()
+    for i in range(28, 0, -1):
+        then = today - datetime.timedelta(days=i)
+        till = then + datetime.timedelta(days=1)
+        op.execute(
+            """
+            SELECT
+                backfill_weekly_report_partitions(
```
Comment (twobraids) [inline alembic/versions/16b2bee7db72_backfill_signature_summary_os_28_days.py]: No need to act: but we end up running the ```backfill_weekly...``` stored procedure 7 times more than necessary.  However, this is only run once and that stored procedure can cope with the unnecessary runs without raising an exception.

**Comment by twobraids** [comment]
r+ if you're satisfied with the data that gets inserted.

**Comment by peterbe** [comment]
What did you mean by the "7 times more"?

File: alembic/versions/16b2bee7db72_backfill_signature_summary_os_28_days.py (hunk 1 above)
Comment (twobraids) [inline alembic/versions/16b2bee7db72_backfill_signature_summary_os_28_days.py]: the partitioning stored procedure, when given a date, will figure find the date of the previous Monday and then create a partition for it.  You run that procedure for each day of the week - effectively telling the procedure to make the partition seven times.  It does make the partition on the first call.  On each of the subsequent six calls, it figures out the date of the Monday and then sees that there already is a partition for that week.  It silently quits at that point, declining to make a partition that already exists.

**Comment by peterbe** [comment]
A ha! So it's benign. Saves me from having to figure out which of all those is a Monday etc.

**Pull request merged**